assets/migrations/postgres/005_drop_indices.sql

```diff
@@ -0,0 +1,46 @@
+-- +goose NO TRANSACTION
+-- This migration updates the primary key of the tuple table, drops old indexes
+
+-- +goose Up
+
+CREATE UNIQUE INDEX CONCURRENTLY idx_tuple_pkey
+    ON tuple (store, object_type, object_id, relation, user_object_type, user_object_id, user_relation);
+
+ALTER TABLE tuple
+    ALTER COLUMN user_object_type SET NOT NULL,
+    ALTER COLUMN user_object_id SET NOT NULL,
+    ALTER COLUMN user_relation SET NOT NULL,
+    DROP CONSTRAINT tuple_pkey,
+    ADD CONSTRAINT tuple_pkey PRIMARY KEY USING INDEX idx_tuple_pkey;
+
+-- +goose StatementBegin
+DROP INDEX CONCURRENTLY idx_tuple_partial_user;
```

Comment:
If we drop these indexes and we don't change our queries to start using the new columns that have been backfilled in the prior migration, then we'll suffer performance penalities. I test this out, and our latency went up because these indexes no longer serve the `ReadUserTuple` and `ReadUsersetTuples` queries, because they use the old column(s) in their query.

Reply:
```sql
Index Scan using idx_reverse_lookup_user on tuple  (cost=0.55..8.58 rows=1 width=121) (actual time=0.026..0.027 rows=1 loops=1)
   Index Cond: ((store = '01H1A9YPAYYH7H54PPPJMVN4J1'::text) AND (object_type = 'document'::text) AND (relation = 'viewer'::text) AND (_user = 'user:0'::text))
   Filter: ((object_id = 'budget'::text) AND (user_type = 'user'::text))
 Planning Time: 0.083 ms
 Execution Time: 0.040 ms
```

versus

```
 Index Scan using idx_tuple_partial_user on tuple  (cost=0.55..8.58 rows=1 width=121) (actual time=0.043..0.044 rows=1 loops=1)
   Index Cond: ((store = '01H1A9YPAYYH7H54PPPJMVN4J1'::text) AND (object_type = 'document'::text) AND (object_id = 'budget'::text) AND (relation = 'viewer'::text) AND (_user = 'user:0'::text))
 Planning Time: 0.292 ms
 Execution Time: 0.054 ms
```